src/interp.rs

```diff
@@ -0,0 +1,1252 @@
+            Some(ConstVal::I64((*a as i64).checked_div(*b as i64)? as u64))
+        }
+        (Operator::I64RemU, [ConstVal::I64(a), ConstVal::I64(b)]) => {
+            Some(ConstVal::I64(a.checked_rem(*b)?))
+        }
+        (Operator::I64RemS, [ConstVal::I64(a), ConstVal::I64(b)]) => {
+            Some(ConstVal::I64((*a as i64).checked_rem(*b as i64)? as u64))
+        }
+        (Operator::I64And, [ConstVal::I64(a), ConstVal::I64(b)]) => Some(ConstVal::I64(a & b)),
+        (Operator::I64Or, [ConstVal::I64(a), ConstVal::I64(b)]) => Some(ConstVal::I64(a | b)),
+        (Operator::I64Xor, [ConstVal::I64(a), ConstVal::I64(b)]) => Some(ConstVal::I64(a ^ b)),
+        (Operator::I64Shl, [ConstVal::I64(a), ConstVal::I64(b)]) => {
+            Some(ConstVal::I64(a.wrapping_shl(*b as u32)))
+        }
+        (Operator::I64ShrS, [ConstVal::I64(a), ConstVal::I64(b)]) => {
+            Some(ConstVal::I64((*a as i64).wrapping_shr(*b as u32) as u64))
+        }
+        (Operator::I64ShrU, [ConstVal::I64(a), ConstVal::I64(b)]) => {
+            Some(ConstVal::I64(a.wrapping_shr(*b as u32)))
+        }
+        (Operator::I64Rotl, [ConstVal::I64(a), ConstVal::I64(b)]) => {
+            Some(ConstVal::I64(a.rotate_left((*b as u32) & 0x3f)))
+        }
+        (Operator::I64Rotr, [ConstVal::I64(a), ConstVal::I64(b)]) => {
+            Some(ConstVal::I64(a.rotate_right((*b as u32) & 0x3f)))
+        }
+
+        (Operator::F32Abs, [ConstVal::F32(a)]) => {
+            Some(ConstVal::F32(f32::from_bits(*a).abs().to_bits()))
+        }
+        (Operator::F32Neg, [ConstVal::F32(a)]) => {
+            Some(ConstVal::F32((-f32::from_bits(*a)).to_bits()))
+        }
+        (Operator::F32Ceil, [ConstVal::F32(a)]) => {
+            Some(ConstVal::F32(f32::from_bits(*a).ceil().to_bits()))
+        }
+        (Operator::F32Floor, [ConstVal::F32(a)]) => {
+            Some(ConstVal::F32(f32::from_bits(*a).floor().to_bits()))
+        }
+        (Operator::F32Trunc, [ConstVal::F32(a)]) => {
+            Some(ConstVal::F32(f32::from_bits(*a).trunc().to_bits()))
+        }
+        (Operator::F32Nearest, [ConstVal::F32(a)]) => {
+            // See
+            // https://github.com/paritytech/wasmi/blob/43ce25d47e26498b9372369345e75dc9632eca8f/crates/core/src/value.rs#L662
+            // for the origin of this algorithm.
+            //
+            // When https://github.com/rust-lang/rust/pull/95317 is
+            // resolved and the resulting API is stable, we can switch
+            // to that instead.
+            let a = f32::from_bits(*a);
+            let round = a.round();
+            let nearest = if a.fract().abs() != 0.5 {
+                round
+            } else {
+                let rem = round % 2.0;
+                if rem == 1.0 {
+                    a.floor()
+                } else if rem == -1.0 {
+                    a.ceil()
+                } else {
+                    round
+                }
+            };
```

Comment:
`round_ties_even` is now stabilized in Rust 1.77, in case we can depend on that version.

Reply:
Done!